lib/puppet/network/http/api.rb

```diff
@@ -6,4 +6,27 @@ def self.not_found
         raise Puppet::Network::HTTP::Error::HTTPNotFoundError.new("No route for #{req.method} #{req.path}", Puppet::Network::HTTP::Issues::HANDLER_NOT_FOUND)
       end)
   end
+
+  def self.master_routes
+    master_prefix = Puppet[:master_url_prefix]
+    if !master_prefix.end_with?("/")
```

Comment:
Do you think maybe we should do this in the setting hook?  Since we already have some slash manipulation going on there?  Or are there cases where we need that setting to not have a trailing slash?